### PR TITLE
 Fix to compilePlayBinaryScala to remove incorrect build cache entry …

### PR DESCRIPTION
**Pull request by devishree90**
…when source files are moved between modules. Issue#5769

Signed-off-by: Devi Sridharan <devishree_90@yahoo.co.in>

### Context
https://github.com/gradle/gradle/issues/5769

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/language-scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java [hunk 1]
```diff
@@ -89,7 +91,12 @@ public CompileOptions getOptions() {
     protected void compile() {
         ScalaJavaJointCompileSpec spec = createSpec();
         configureIncrementalCompilation(spec);
-        getCompiler(spec).execute(spec);
+        Compiler<ScalaJavaJointCompileSpec> compiler = getCompiler(spec);
+        File analysisFile = getScalaCompileOptions().getIncrementalOptions().getAnalysisFile();
+        if(analysisFile==null){
```
Comment (wolfs) [inline subprojects/language-scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java]: I guess you need to check if the file exists - not if the location of the analysis file is configured.

The location of the analysis file is configured by default.

**Comment by wolfs** [comment]
@devishree90 Does the PR now fix #5769?

**Comment by devishree90** [comment]
@wolfs - No. I still see incorrect build files(Duplicate entry)

**Comment by wolfs** [comment]
@devishree90 I think I fixed the issue. See https://github.com/gradle/gradle/pull/6093.

**Pull request closed without merging**